cmd/gapit/trace2apk.go

```diff
@@ -0,0 +1,229 @@
+// Copyright (C) 2019 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"flag"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/google/gapid/core/app"
+	"github.com/google/gapid/core/app/layout"
+	"github.com/google/gapid/core/log"
+	"github.com/google/gapid/gapis/service"
+)
+
+type trace2apkVerb struct{ Trace2apkFlags }
+
+func init() {
+	verb := &trace2apkVerb{}
+	app.AddVerb(&app.Verb{
+		Name:      "trace2apk",
+		ShortHelp: "Create an stand-alone APK which replays a trace (off-screen)",
+		Action:    verb,
+	})
+}
+
+func (verb *trace2apkVerb) Run(ctx context.Context, flags flag.FlagSet) error {
+	if flags.NArg() != 1 {
+		app.Usage(ctx, "Exactly one gfx trace file expected, got %d", flags.NArg())
+		return nil
+	}
+
+	client, capturePath, err := getGapisAndLoadCapture(ctx, verb.Gapis, verb.Gapir, flags.Arg(0), verb.CaptureFileFlags)
+	if err != nil {
+		return err
+	}
+	defer client.Close()
+
+	if err := client.ExportReplay(ctx, capturePath, nil, "assets/replay_export", &service.ExportReplayOptions{}); err != nil {
+		return log.Err(ctx, err, "Failed to export replay")
+	}
+
+	// TODO: extract in common.go
+	boxedCapture, err := client.Get(ctx, capturePath.Path(), nil)
+	if err != nil {
+		return log.Err(ctx, err, "Failed to load the capture")
+	}
+	capture := boxedCapture.(*service.Capture)
+
+	refAPKPath, err := layout.GapidApk(ctx, capture.ABI)
+	if err != nil {
+		return err
+	}
+
+	trace := filepath.Base(flags.Arg(0))
+	replayLabel := strings.TrimSuffix(trace, ".gfxtrace")
+	replayPackage := replayLabel + ".replay"
+	replayAPK := replayPackage + ".apk"
+	log.I(ctx, "Replay apk: %s", replayAPK)
+
+	// Extract reference APK
+	if err := execCommand(ctx, "unzip", "-o", refAPKPath.String()); err != nil {
+		return err
+	}
+
+	// Create new manifest
+	f, err := os.Create("NewManifest.xml")
+	if err != nil {
+		return err
+	}
+	t, err := template.New("manifest").Parse(manifest())
+	if err != nil {
+		return err
+	}
+	type ManifestEntries struct {
+		Label   string
+		Package string
+	}
+	var entries = ManifestEntries{replayLabel, replayPackage}
+	t.Execute(f, entries)
+	f.Close()
+
+	// Re-assemble new APK with the new manifest
+	if err := execCommand(ctx, "aapt2", "link", "-o", replayAPK, "--manifest", "NewManifest.xml", "-I", os.ExpandEnv("${ANDROID_SDK_HOME}/platforms/android-26/android.jar")); err != nil {
```

Comment:
It may make sense to use the aapt2 from the Android SDK. That way more users will have it work.

Reply:
`${ANDROID_SDK_HOME}/platforms/android-26/android.jar` This probably won't work on Windows.

We can `path.join(getenv(ANDROID_SDK_HOME),  ...)`
Also it would be a good idea allow the user specify an sdk directory that isn't ANDROID_SDK_HOME. Forcing env_vars is not nice.